### ISSUE TITLE
What does it use to see who is online?

### ISSUE THREAD

**Issue by haylinmoore** [joker314/PrivateOnline#2]

**Comment by haylinmoore** [comment]
From what I gather it posts in a forum topic, but I don't think scratch would allow it

**Comment by towerofnix** [comment]
Yup, that's what it does. Such is a significant part of the reason that this project is abandoned. (You get a dilemma of sorts - either you rely on an external server, which could track what you're doing, or you use Scratch's APIs, but in that case you're really *ab*using them, which is bad, too.)

**Comment by haylinmoore** [comment]
Why not have the external server run on heroku that is linked with a git repo, so the backend is forced to be open

**Comment by towerofnix** [comment]
That's a pretty good idea, actually.

What if *you* did this, for the original isOnline project you and the rest of the iO team are working on, making the backend open as you described? :)

**Comment by haylinmoore** [comment]
Here is the issue. Heroku, can be slow, and takes 30 seconds to boot up. Also is it possible to have a constant database ever after rebuilds?

**Comment by haylinmoore** [comment]
But if you find a way, that also lets us keep our user key privates I will happlily make the switch

**Comment by towerofnix** [comment]
>Here is the issue. Heroku, can be slow, and takes 30 seconds to boot up.

Test on a local server; publish the extension so that it's connected to the Heroku server. Since the Heroku server will only reboot once every publish, that 30 seconds doesn't matter so much. And in development, you'll use a faster server.

>Also is it possible to have a constant database ever after rebuilds?

I don't know - you proposed this :)

But you can probably find information about that online.

**Comment by haylinmoore** [comment]
Do you feel it would be ok if the database was stored on our servers? but is run on heroku

**Comment by towerofnix** [comment]
That's better than having the code also be run on your servers, but it isn't perfect, because, with the database on your servers, you can keep a track just like before. After all, Heroku has to make HTTP requests to your database server, and you *could* be running a script which watches for HTTP requests on that server. That script would keep a record of the database fields being set; so, a running count of who visits profiles the most. So really, it doesn't solve the problem there was with isOnline.

**Comment by haylinmoore** [comment]
I don't see why knowing what profiles are popular is *bad*, like "Oh, no, lots of people go to griffpatch, damn that's unexpected"

**Comment by haylinmoore** [comment]
Anyways say we did, what good would come from that? Like we would not sell, if anything it would only show "Wow, people go to the pages of popular scratchers"
